### PR TITLE
51 search implementation

### LINKED ISSUE #108: Optimize search code
Simplify this pls.
 
 ```
searchProducts.forEach((category) => {
    document.querySelector(`#search-dialog .search-results-container .${category}-category-list`).innerHTML = ""
 });
```

https://github.com/hlxsites/fonterra-fernleaf/pull/101#discussion_r1290416667

_Originally posted by @sselvara in https://github.com/hlxsites/fonterra-fernleaf/pull/101#discussion_r1290410615_

### PR DESCRIPTION
**Pull request by noormd08**
Please always provide the [GitHub issue(s)](../issues) your PR is for, as well as test URLs where your change can be observed (before and after):

Fix #108

Test URLs:
- Before: https://main--fonterra-fernleaf--hlxsites.hlx.page/
- After: https://51-search-implementation--fonterra-fernleaf--hlxsites.hlx.page/en

### CODE REVIEW

File: blocks/header/search.js [hunk 1]
```diff
@@ -49,20 +47,22 @@ const addProductsHTML = (categoryName, productList, placeholders) => {
...
+        <span class='split-bar'></span>
+        <a href='${productPlaceholder.viewLink}'>${productPlaceholder.viewText}</a>
+      </div>
     </div>
     <div class='product-list-results'>
-      ${productList.map((item) =>
-    getProductListHTML(categoryName, item)).join('')}
+      ${productList.map((item) => getProductListHTML(categoryName, item)).join('')}
     </div>
+    ${categoryName.toLowerCase() !== 'story' ? `
```
Comment (sselvara) [inline blocks/header/search.js:61]: can you keep 'story' in constant?

File: blocks/header/search.js [hunk 2]
```diff
@@ -94,10 +94,7 @@ export async function performSearch(value) {
     if (placeholders && searchData) {
       let filteredResults = [];
       productCategories.map((category) => {
-        filteredResults = searchData.filter((el) =>
-          /* eslint implicit-arrow-linebreak: ["error", "below"] */
-          /* eslint max-len: ["error", { "code": 300 }] */
-          el.category.toLowerCase() === category.toLowerCase() && (el.title.toLowerCase().includes(searchValue) || el.shorttitle.toLowerCase().includes(searchValue) || el.description.toLowerCase().includes(searchValue) || el.tags.toLowerCase().includes(searchValue)));
+        filteredResults = searchData.filter((el) => el.category.toLowerCase() === category.toLowerCase() && ['title', 'shorttitle', 'description', 'tags'].some((field) => el[field].toLowerCase().includes(searchValue.toLowerCase())));
         if (category !== 'Story') {
```
Comment (sselvara) [inline blocks/header/search.js:98]: use the same comparison as in line no: 98, ${categoryName.toLowerCase() !== 'story'

**Review by sselvara — CHANGES_REQUESTED** [review]

File: blocks/header/search.js (hunk 2 above)
Comment (noormd08) [reply blocks/header/search.js]: Added index instead of direct usage.

File: blocks/header/search.js (hunk 1 above)
Comment (noormd08) [reply blocks/header/search.js]: Resolved

**Review by sselvara — APPROVED** [review]

**Pull request merged**